src/sql/base/browser/ui/table/formatters.ts

```diff
@@ -44,7 +44,11 @@ export function textFormatter(row: number | undefined, cell: any | undefined, va
 	let valueToDisplay = '';
 	let titleValue = '';
 
-	if (DBCellValue.isDBCellValue(value)) {
+	if (value === null) {
+		value = { displayValue: 'NULL', ariaLabel: 'NULL', isNull: true };
+	}
+
+	if (value !== null && DBCellValue.isDBCellValue(value)) {
```

Comment:
value can never be null here (you set it on line 48 if it's null)

Also is there a reason you're checking for null in these changes versus undefined? Most of our code uses undefined instead of null

Reply:
attempting to use "undefined" instead of null causes the entry box to show [Object object] instead of an empty space (as intended, because of the grid interface) Also the use of null instead of undefined is used to differentiate from loading the header bars over a null cell (from the very start when the grid is being initialized). The reason for the null check on line 48 is because the current value of the cell is passed back for text formatting upon hitting entering and the value in the cell is null

Reply:
There are a few reasons for the use of Null here:

Attempting to use undefined instead of null causes the entry box to show [Object object] instead of an empty space (because of the grid interface presumably only recognizing null as empty)

Also the use of null instead of undefined is used to differentiate from loading the header bars over a null cell (from the very start when the grid is being initialized).

The reason for the null check on line 48 is because the entered value of the cell is directly passed back for text formatting upon hitting "enter" and usually it is a string (which the value in the cell is changed into it from my understanding). We cannot pass in the string "null" or DBCell null from editData.component otherwise that will result in the cell value being changed to an object instead of actual null.

Reply:
yes, let's please try to use `undefined` in all our code unless we explicitly need `null` for some reason.  Thanks!